src/abi.h

```diff
@@ -367,4 +368,29 @@ struct netplan_net_definition {
     /* netplan-feature: vrf */
     NetplanNetDefinition* vrf_link;
     guint vrf_table;
+
+    /* TODO: `link` can also be used for vrf/vlan/sriov/... in the future */
+    NetplanNetDefinition* link;
+    NetplanTristate bridge_neigh_suppress;
+
+    /* vxlan */
+    /* netplan-feature: vxlan */
+    gboolean has_vxlans;
+    struct {
+        guint vni;
+        guint ageing;
+        guint limit;
+        guint tos;
+        guint flow_label;
+        guint source_port_min;
+        guint source_port_max;
+        gboolean mac_learning;
+        gboolean arp_proxy;
+        gboolean short_circuit;
+        gboolean independent;
+        NetplanFlags notifications;
+        NetplanFlags checksums;
+        NetplanFlags extensions;
+        NetplanTristate do_not_fragment;
+    } vxlan;
```

Comment:
suggestion: how about using a private struct separately allocated?

That way we avoid exposing even more data through the legacy ABI, and non-vxlan netdefs don't have to pay the full price of this relatively big new addition.

```c
typedef struct netplan_vxlan NetplanVxlan;
struct netplan_net_definition {
    // ...
    NetplanVxlan* vxlan;
};

// In types.h
struct netplan_vxlan {
        guint vni;
        guint ageing;
        guint limit;
        guint tos;
        guint flow_label;
        guint source_port_min;
        guint source_port_max;
        gboolean mac_learning;
        gboolean arp_proxy;
        gboolean short_circuit;
        gboolean independent;
        NetplanFlags notifications;
        NetplanFlags checksums;
        NetplanFlags extensions;
        NetplanTristate do_not_fragment;
};
```

Of course there'd need to be some more changes around to account for the new redirection, as well as dealing with allocaitng and freeing the struct.

Reply:
This is quite a rework.. But you're right it makes sense not pulling it into the legacy ABI. Luckily the tests are all there already, so we can easily refactor this!